tests/test_client.py

```diff
@@ -0,0 +1,110 @@
+import unittest
+
+from wideq.client import (
+    BitValue, EnumValue, ModelInfo, RangeValue, ReferenceValue)
+
+
+DATA = {
+    'Value': {
+        'AntiBacterial': {
+            'default': '0',
+            'label': '@WM_DRY27_BUTTON_ANTI_BACTERIAL_W',
+            'option': {
+                '0': '@CP_OFF_EN_W',
+                '1': '@CP_ON_EN_W'
+            },
+            'type': 'Enum'
+        },
+        'Course': {
+            'option': ['Course'],
+            'type': 'Reference',
+        },
+        'Initial_Time_H': {
+            'default': 0,
+            'option': {'max': 24, 'min': 0},
+            'type': 'Range'
+        },
+        'Option1': {
+            'default': '0',
+            'option': [
+                {
+                    'default': '0',
+                    'length': 1,
+                    'startbit': 0,
+                    'value': 'ChildLock'
+                },
+                {
+                    'default': '0',
+                    'length': 1,
+                    'startbit': 1,
+                    'value': 'ReduceStatic'
+                },
+                {
+                    'default': '0',
+                    'length': 1,
+                    'startbit': 2,
+                    'value': 'EasyIron'
+                },
+                {
+                    'default': '0',
+                    'length': 1,
+                    'startbit': 3,
+                    'value': 'DampDrySingal'
+                },
+                {
+                    'default': '0',
+                    'length': 1,
+                    'startbit': 4,
+                    'value': 'WrinkleCare'
+                },
+                {
+                    'default': '0',
+                    'length': 1,
+                    'startbit': 7,
+                    'value': 'AntiBacterial'
+                }
+            ],
+            'type': 'Bit'
+        },
+        'Unexpected': {'type': 'Unexpected'},
+    },
+}
+
+
+class ModelInfoTest(unittest.TestCase):
+
+    def setUp(self):
+        super().setUp()
+        self.model_info = ModelInfo(DATA)
+
+    def test_value_enum(self):
+        actual = self.model_info.value('AntiBacterial')
+        expected = EnumValue({'0': '@CP_OFF_EN_W', '1': '@CP_ON_EN_W'})
+        self.assertEqual(expected, actual)
+
+    def test_value_range(self):
+        actual = self.model_info.value('Initial_Time_H')
+        expected = RangeValue(min=0, max=24, step=1)
+        self.assertEqual(expected, actual)
+
+    def test_value_bit(self):
+        actual = self.model_info.value('Option1')
+        expected = BitValue({
+            0: {'length': 1, 'value': 'ChildLock'},
+            1: {'length': 1, 'value': 'ReduceStatic'},
+            2: {'length': 1, 'value': 'EasyIron'},
+            3: {'length': 1, 'value': 'DampDrySingal'},
+            4: {'length': 1, 'value': 'WrinkleCare'},
+            7: {'length': 1, 'value': 'AntiBacterial'},
```

Comment:
Just out of curiosity, does `length` always seem to be 1 in your experience? The name "bit" for the value type seems to suggest that. If so, maybe we should drop the `length` field and just keep track of the value name?

It wouldn't be the first time something in this API is over-engineered and only actually used in a specific, simpler special case…

Reply:
There is at least once case where it's something other than 1, but I'm pretty sure the library wouldn't make use of this value, so it's probably suffice to just keep the name.

```json
{ "RinseOption": {
          "type": "Bit",
          "default": "0",
          "option": [
            {
              "startbit": 0,
              "length": 4,
              "default": "0",
              "value": "RinseCount"
            },
            {
              "startbit": 4,
              "length": 4,
              "default": "0",
              "value": "ExtraRinseCount"
            }
          ]
        }
}
```